### PR TITLE
Change to new Treesitter highlight groups

### LINKED ISSUE #15: Use nvim-treesitter's new highlighting groups
Hello,

I just tried to install the everblush colorscheme for Neovim. However, the colors look completely different from the preview image in the README, as you can see below.

The issue seems to be related to the following breaking change in nvim-treesitter: https://github.com/nvim-treesitter/nvim-treesitter/issues/2293#issuecomment-1279974776
The "TS" highlighting groups have been rendered obsolete, and this is what causes the issue for me.

I have solved the issue for my own config, using this temporary fix I found online: https://gist.github.com/rockerBOO/be0770242db9b7215b8e83e135516a65
However, this is more of a band-aid rather than a true fix!

### PR DESCRIPTION
**Pull request by diniamo**
Closes #15

### CODE REVIEW

File: lua/everblush/config.lua [hunk 1]
```diff
@@ -102,64 +102,65 @@ M.highlights_base = function (colors)
...
+    ["@constant"] = { fg = colors.color6 },
+    ["@constant.builtin"] = { fg = colors.color4 },
+    ["@constant.macro"] = { fg = colors.color3 },
+    ["@constructor"] = { fg = colors.color4 },
+    ["@exception"] = { fg = colors.color8 },
+    ["@field"] = { fg = colors.color1 },
+    ["@float"] = { fg = colors.color8 },
+    ["@function"] = { fg = colors.color1 },
+    ["@func.builtin"] = { fg = colors.color14 },
+    ["@func.macro"] = { fg = colors.color2 },
```
Comment (karimudev) [inline lua/everblush/config.lua]: The highlighting groups are "@function.builtin" and "@function.macro"

File: lua/everblush/config.lua [hunk 2]
```diff
@@ -102,64 +102,65 @@ M.highlights_base = function (colors)
...
+    ["@include"] = { fg = colors.color9 },
+	["@operator"] = { fg = colors.color7 },
```
Comment (karimudev) [inline lua/everblush/config.lua]: Small formatting issue with the extra tab?

File: lua/everblush/config.lua [hunk 3]
```diff
@@ -102,64 +102,65 @@ M.highlights_base = function (colors)
...
+    ["@keyword"] = { fg = colors.color5 },
+    ["@keyword.function"] = { fg = colors.color4 },
+    ["@keyword.operator"] = { fg = colors.color12 },
+    ["@keyword.return"] = { fg = colors.color4 },
+    ["@label"] = { fg = colors.color4 },
+    ["@method"] = { fg = colors.color12 },
+    ["@namespace"] = { fg = colors.color9 },
+    ["@number"] = { fg = colors.color3 },
+    ["@parameter"] = { fg = colors.color1 },
+    ["@parameter.reference"] = { fg = colors.color9 },
```
Comment (karimudev) [inline lua/everblush/config.lua:131]: "@parameter.reference" doesn't seem to exist according to this: https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md

"@reference" and "@text.reference" do exist though, but I am not sure of which one is the correct one.

File: lua/everblush/config.lua [hunk 4]
```diff
@@ -102,64 +102,65 @@ M.highlights_base = function (colors)
...
+    ["@tag.delimiter"] = { fg = colors.color7 },
+    ["@text"] = { fg = colors.color7 },
+    ["@text.strong"] = { fg = colors.color7, bold = true },
+    ["@text.emphasis"] = { fg = colors.color7, italic = true },
+    ["@text.underline"] = { fg = colors.color5, underline = true },
+    ["@text.strike"] = { fg = colors.color7, strikethrough = true },
+    ["@text.title"] = { fg = colors.color3, bold = true },
+    ["@text.literal"] = { fg = colors.color2, italic = true },
+    ["@text.uri"] = { fg = colors.color3, underline = true },
+    ["@math"] = { fg = colors.color6 },
```
Comment (karimudev) [inline lua/everblush/config.lua]: Should be "@text.math"

File: lua/everblush/config.lua [hunk 5]
```diff
@@ -102,64 +102,65 @@ M.highlights_base = function (colors)
...
+    ["@text.reference"] = { fg = colors.color6 },
+    ["@environtment"] = { fg = colors.color5 },
+    ["@environment.name"] = { fg = colors.color3 },
```
Comment (karimudev) [inline lua/everblush/config.lua]: Should be "@text.environment" and "@text.environment.name"

**Comment by karimudev** [comment]
Apologies for the whole list of comments, it's my first time using this Github tool and I didn't know that each comment would be displayed individually instead of all as one review!

The PR works on my machine, but there are just a few highlighting groups that are not named correctly. I think I have pointed them all out.

**Comment by diniamo** [comment]
No, problem, I'll have all of it fixed by the end of tomorrow. It would be great if you could figure out the reference one.

File: lua/everblush/config.lua (hunk 3 above)
Comment (karimudev) [reply lua/everblush/config.lua:131]: Did some more digging. First of all, "@reference" is not a group used for highlighting. "@text.reference" is, but it is used for footnotes, citations, etc. so I doubt.

I cannot find anything recent about this "@parameter.reference" (or TSParameterReference) that is recent. Maybe this highlighting group does not exist anymore and can be deleted? Sorry, I'm not familiar with nvim-treesitter and I cannot find more documentation, so I cannot provide more info

Comment (diniamo) [reply lua/everblush/config.lua:131]: That's already plenty of information, thanks. I'll probably end up deleting it after asking in nvim-dev.

Comment (diniamo) [reply lua/everblush/config.lua:131]: Hmm I didn't get a clear answer, but I think we should keep it

**Comment by mngshm** [comment]
Ahh damn, btw @C0RNPOP thanks for the help with it.

File: lua/everblush/config.lua (hunk 3 above)
Comment (diniamo) [reply lua/everblush/config.lua:131]: It's staying.

File: lua/everblush/config.lua (hunk 2 above)
Comment (diniamo) [reply lua/everblush/config.lua]: I have no clue what's causing this, it looks fine locally.

**Comment by diniamo** [comment]
OK I think I fixed everything, will merge right after @C0RNPOP confirms it.

**Comment by karimudev** [comment]
Hello,

Everything looks good to me, and it works on my machine. Should be good to merge!

**Pull request merged**